sdk/src/main/java/tech/dojo/pay/sdk/payemntintent/PaymentIntentProvider.kt

```diff
@@ -6,18 +6,21 @@ import kotlinx.coroutines.launch
 import tech.dojo.pay.sdk.DojoPaymentIntentResult
 import tech.dojo.pay.sdk.payemntintent.data.PaymentIntentApiBuilder
 import tech.dojo.pay.sdk.payemntintent.data.PaymentIntentRepository
+
 @Suppress("SwallowedException")
 internal class PaymentIntentProvider(
     private val paymentIntentRepository: PaymentIntentRepository = PaymentIntentRepository(
-        PaymentIntentApiBuilder().create()
-    )
+        PaymentIntentApiBuilder().create(),
+    ),
+    private val repositoryScope: CoroutineScope = CoroutineScope(Dispatchers.IO),
```

Comment:
Not sure if this is a pattern but it is normally nicer to setup the class as a CoroutineScope and then set a Dispatchers to is scope 
A bit like this https://github.com/dojo-engineering/superpay-android/blob/36db9b76070cf230f1faa6bed95da63688329b35/feature-transactions/transactions-presentation/src/main/java/uk/co/paymentsense/superpay/transactions/presentation/TransactionPollingHandler.kt#L25-L30